? arm/dts/sun8i-v3s-licheepi-zero.dts
? arm/dts/sun8i-v3s.dtsi
? arm/sunxi/sun8i_v3s_gpio.c
Index: arm/sunxi/files.sunxi
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/files.sunxi,v
retrieving revision 1.69
diff -u -r1.69 files.sunxi
--- arm/sunxi/files.sunxi	5 May 2021 10:24:04 -0000	1.69
+++ arm/sunxi/files.sunxi	12 Jun 2021 02:58:25 -0000
@@ -136,6 +136,7 @@
 file	arch/arm/sunxi/sun7i_a20_gpio.c		sunxi_gpio & soc_sun7i_a20
 file	arch/arm/sunxi/sun8i_a83t_gpio.c	sunxi_gpio & soc_sun8i_a83t
 file	arch/arm/sunxi/sun8i_h3_gpio.c		sunxi_gpio & soc_sun8i_h3
+file	arch/arm/sunxi/sun8i_v3s_gpio.c		sunxi_gpio & soc_sun8i_v3s
 file	arch/arm/sunxi/sun9i_a80_gpio.c		sunxi_gpio & soc_sun9i_a80
 file	arch/arm/sunxi/sun50i_a64_gpio.c	sunxi_gpio & soc_sun50i_a64
 file	arch/arm/sunxi/sun50i_h6_gpio.c		sunxi_gpio & soc_sun50i_h6
Index: arm/sunxi/sunxi_de2_ccu.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_de2_ccu.c,v
retrieving revision 1.7
diff -u -r1.7 sunxi_de2_ccu.c
--- arm/sunxi/sunxi_de2_ccu.c	27 Jan 2021 03:10:20 -0000	1.7
+++ arm/sunxi/sunxi_de2_ccu.c	12 Jun 2021 02:58:25 -0000
@@ -100,6 +100,8 @@
 static const struct device_compatible_entry compat_data[] = {
 	{ .compat = "allwinner,sun8i-h3-de2-clk",
 	  .data = &sun8i_h3_de2_config },
+	{ .compat = "allwinner,sun8i-v3s-de2-clk",
+	  .data = &sun8i_h3_de2_config },
 	{ .compat = "allwinner,sun50i-a64-de2-clk",
 	  .data = &sun50i_a64_de2_config },
 	{ .compat = "allwinner,sun50i-h5-de2-clk",
Index: arm/sunxi/sunxi_drm.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_drm.c,v
retrieving revision 1.14
diff -u -r1.14 sunxi_drm.c
--- arm/sunxi/sunxi_drm.c	24 Apr 2021 23:36:28 -0000	1.14
+++ arm/sunxi/sunxi_drm.c	12 Jun 2021 02:58:25 -0000
@@ -66,6 +66,7 @@
 
 static const struct device_compatible_entry compat_data[] = {
 	{ .compat = "allwinner,sun8i-h3-display-engine" },
+	{ .compat = "allwinner,sun8i-v3s-display-engine" },
 	{ .compat = "allwinner,sun50i-a64-display-engine" },
 	DEVICE_COMPAT_EOL
 };
Index: arm/sunxi/sunxi_gpio.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_gpio.c,v
retrieving revision 1.36
diff -u -r1.36 sunxi_gpio.c
--- arm/sunxi/sunxi_gpio.c	24 Apr 2021 23:36:28 -0000	1.36
+++ arm/sunxi/sunxi_gpio.c	12 Jun 2021 02:58:25 -0000
@@ -111,6 +111,10 @@
 	{ .compat = "allwinner,sun8i-h3-r-pinctrl",
 	  .data = &sun8i_h3_r_padconf },
 #endif
+#ifdef SOC_SUN8I_V3S
+	{ .compat = "allwinner,sun8i-v3s-pinctrl",
+	  .data = &sun8i_v3s_padconf },
+#endif
 #ifdef SOC_SUN9I_A80
 	{ .compat = "allwinner,sun9i-a80-pinctrl",
 	  .data = &sun9i_a80_padconf },
Index: arm/sunxi/sunxi_gpio.h
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_gpio.h,v
retrieving revision 1.9
diff -u -r1.9 sunxi_gpio.h
--- arm/sunxi/sunxi_gpio.h	27 May 2019 23:27:14 -0000	1.9
+++ arm/sunxi/sunxi_gpio.h	12 Jun 2021 02:58:25 -0000
@@ -79,6 +79,10 @@
 extern const struct sunxi_gpio_padconf sun8i_h3_r_padconf;
 #endif
 
+#ifdef SOC_SUN8I_V3S
+extern const struct sunxi_gpio_padconf sun8i_v3s_padconf;
+#endif
+
 #ifdef SOC_SUN9I_A80
 extern const struct sunxi_gpio_padconf sun9i_a80_padconf;
 extern const struct sunxi_gpio_padconf sun9i_a80_r_padconf;
Index: arm/sunxi/sunxi_lcdc.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_lcdc.c,v
retrieving revision 1.12
diff -u -r1.12 sunxi_lcdc.c
--- arm/sunxi/sunxi_lcdc.c	27 Jan 2021 03:10:20 -0000	1.12
+++ arm/sunxi/sunxi_lcdc.c	12 Jun 2021 02:58:25 -0000
@@ -108,6 +108,7 @@
 
 static const struct device_compatible_entry compat_data[] = {
 	{ .compat = "allwinner,sun8i-h3-tcon-tv",	.value = TYPE_TCON1 },
+	{ .compat = "allwinner,sun8i-v3s-tcon",		.value = TYPE_TCON0 },
 	{ .compat = "allwinner,sun50i-a64-tcon-lcd",	.value = TYPE_TCON0 },
 	{ .compat = "allwinner,sun50i-a64-tcon-tv",	.value = TYPE_TCON1 },
 	DEVICE_COMPAT_EOL
Index: arm/sunxi/sunxi_mixer.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_mixer.c,v
retrieving revision 1.16
diff -u -r1.16 sunxi_mixer.c
--- arm/sunxi/sunxi_mixer.c	27 Jan 2021 03:10:20 -0000	1.16
+++ arm/sunxi/sunxi_mixer.c	12 Jun 2021 02:58:25 -0000
@@ -201,6 +201,8 @@
 static const struct device_compatible_entry compat_data[] = {
 	{ .compat = "allwinner,sun8i-h3-de2-mixer-0",
 	  .data = &mixer0_data },
+	{ .compat = "allwinner,sun8i-v3s-de2-mixer",
+	  .data = &mixer0_data },
 	{ .compat = "allwinner,sun50i-a64-de2-mixer-0",
 	  .data = &mixer0_data },
 	{ .compat = "allwinner,sun50i-a64-de2-mixer-1",
Index: arm/sunxi/sunxi_platform.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_platform.c,v
retrieving revision 1.43
diff -u -r1.43 sunxi_platform.c
--- arm/sunxi/sunxi_platform.c	24 Apr 2021 23:36:28 -0000	1.43
+++ arm/sunxi/sunxi_platform.c	12 Jun 2021 02:58:25 -0000
@@ -469,6 +469,7 @@
 
 ARM_PLATFORM(sun8i_h2plus, "allwinner,sun8i-h2-plus", &sun8i_platform);
 ARM_PLATFORM(sun8i_h3, "allwinner,sun8i-h3", &sun8i_platform);
+ARM_PLATFORM(sun8i_v3s, "allwinner,sun8i-v3s", &sun8i_platform);
 
 static const struct arm_platform sun8i_a83t_platform = {
 	.ap_devmap = sun8i_a83t_platform_devmap,
Index: arm/sunxi/sunxi_sramc.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_sramc.c,v
retrieving revision 1.11
diff -u -r1.11 sunxi_sramc.c
--- arm/sunxi/sunxi_sramc.c	27 Jan 2021 03:10:20 -0000	1.11
+++ arm/sunxi/sunxi_sramc.c	12 Jun 2021 02:58:25 -0000
@@ -47,6 +47,7 @@
 	{ .compat = "allwinner,sun4i-a10-sram-controller" },
 	{ .compat = "allwinner,sun4i-a10-system-control" },
 	{ .compat = "allwinner,sun8i-h3-system-control" },
+	{ .compat = "allwinner,sun8i-v3s-system-control" },
 	{ .compat = "allwinner,sun50i-a64-system-control" },
 	{ .compat = "allwinner,sun50i-h5-system-control" },
 	{ .compat = "allwinner,sun50i-h6-system-control" },
Index: arm/sunxi/sunxi_usbphy.c
===================================================================
RCS file: /cvsroot/src/sys/arch/arm/sunxi/sunxi_usbphy.c,v
retrieving revision 1.16
diff -u -r1.16 sunxi_usbphy.c
--- arm/sunxi/sunxi_usbphy.c	27 Jan 2021 03:10:20 -0000	1.16
+++ arm/sunxi/sunxi_usbphy.c	12 Jun 2021 02:58:25 -0000
@@ -82,6 +82,7 @@
 	USBPHY_A83T,
 	USBPHY_H3,
 	USBPHY_H6,
+	USBPHY_V3S,
 };
 
 static const struct device_compatible_entry compat_data[] = {
@@ -91,6 +92,7 @@
 	{ .compat = "allwinner,sun7i-a20-usb-phy",	.value = USBPHY_A20 },
 	{ .compat = "allwinner,sun8i-a83t-usb-phy",	.value = USBPHY_A83T },
 	{ .compat = "allwinner,sun8i-h3-usb-phy",	.value = USBPHY_H3 },
+	{ .compat = "allwinner,sun8i-v3s-usb-phy",	.value = USBPHY_V3S },
 	{ .compat = "allwinner,sun50i-a64-usb-phy",	.value = USBPHY_A64 },
 	{ .compat = "allwinner,sun50i-h6-usb-phy",	.value = USBPHY_H6 },
 	DEVICE_COMPAT_EOL
Index: evbarm/conf/GENERIC
===================================================================
RCS file: /cvsroot/src/sys/arch/evbarm/conf/GENERIC,v
retrieving revision 1.97
diff -u -r1.97 GENERIC
--- evbarm/conf/GENERIC	5 May 2021 10:24:04 -0000	1.97
+++ evbarm/conf/GENERIC	12 Jun 2021 02:58:25 -0000
@@ -25,6 +25,7 @@
 options 	SOC_SUN7I_A20
 options 	SOC_SUN8I_A83T
 options 	SOC_SUN8I_H3
+options 	SOC_SUN8I_V3S
 options 	SOC_SUN9I_A80
 options 	SOC_TEGRA124
 options 	SOC_VIRT
